### ISSUE TITLE
Sorting fails when a one or more sorted columns is hidden

### ISSUE THREAD

**Issue by LordGalahad** [6pac/SlickGrid#1015]
### Describe the bug

I'm not sure if this is intended behavior though.

When I have multiColumnSort enabled, and I have multiple columns currently selected like this:

![image](https://github.com/6pac/SlickGrid/assets/61867299/d9aa1361-48eb-44fa-8fdf-6ddfbf11b377)

And then try to hide one of the sorted columns (like GBP), so it ends up looking like this:

![image](https://github.com/6pac/SlickGrid/assets/61867299/33121160-701b-4e24-8775-2cb8e1198a68)

And then I try to click on another column (like vNoise), I get this error:

![image](https://github.com/6pac/SlickGrid/assets/61867299/31e2ed93-9135-4893-a339-d5672190565a)

![image](https://github.com/6pac/SlickGrid/assets/61867299/3b942731-8209-44c2-82c7-a615fe94d373)

We had no problems with the old slickgrid. But since this is new implementation, it may seem to be limitation?

### Reproduction

1. Set multiColumnSort to true
2. Select  two or more columns to be sorted
3. Hide one of the sorted columns
4. Click on a different column for sorting

### Which Framework are you using?

Other

### Environment Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 3.39 GB / 15.73 GB
  Binaries:
    Node: 18.18.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.15.6 - ~\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Edge: Chromium (124.0.2478.51)
    Internet Explorer: 11.0.19041.3636
  Slickgrid: 5.8.0
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/6pac/SlickGrid/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [Wikis](https://github.com/6pac/SlickGrid/wiki).
- [X] Check that there isn't [already an issue](https://github.com/6pac/SlickGrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/6pac/SlickGrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
There's not enough info/details to help you, just doing print screens of UI and code is not helpful enough for us to understand the problem. I'm not even sure if you're asking about the new `hidden` property on the column definition? If that is the case then @6pac could help since I'm not using that option on my side

**Comment by LordGalahad** [comment]
I actually tried it on one of the samples - https://6pac.github.io/SlickGrid/examples/example-column-hidden.html

I tried setting options.multiColumnSort to true. Then I select two columns to be sorted. Then I hide one of the two sorted columns. Then I select a different column to sort. Then I get the exception error

Here is a loom video: https://www.loom.com/share/ab26e2aa6e674a2d93cc0428da2a6b2d

**Comment by 6pac** [comment]
@LordGalahad thanks, too busy today, but I'll have a look tomorrow morning. There was a lot of debate around the changes, I'll have to go back and look at exactly how we did it.

**Comment by 6pac** [comment]
Sorry, I'm trying to repro but I can't get the error to happen in the examples with a .js source. I notice you are referencing a .ts source directly in your examples file. How are you doing that?  
These might seem a silly question, but this getting dangerously close to being only @ghiscoding 's project.  I don't use TS or any of the tooling that the current repo uses. At the moment they are a significant barrier to me being able to diagnose or patch anything. I do all my diagnosis and coding in js and then have to back-patch to ts and get the dist files to flow through.

**Comment by 6pac** [comment]
@LordGalahad thanks, I could repro. I was missing the final shift-click.

@ghiscoding this is actually nothing to do with the ```hidden``` property - it's happening because the grid menu is *not* using the hidden property but it removing the column from the ```columns``` array. This must have been a pre-existing bug. I have created a PR to fix. (note - bugs like this are exactly why we needed the ```hidden``` property - keeping multiple column lists in sync across possibly multiple components is a nightmare)

Both of you: this does raise another usage question. If we are ordering by a column and it is hidden, should it be removed from the sort list? I can see use cases either way. Should we create an option, say ```removeHiddenColsFromSort```, defaulting to false so as to be non breaking? This would require additional changes, which I can do.

**Comment by ghiscoding** [comment]
@6pac that bug was also reported in the past when you introduced the hidden feature, see #546 
I didn't look at the code, I just remembered that it was a similar bug

**Comment by 6pac** [comment]
@ghiscoding ah yes, good catch. Any opinion on the usage question?

**Comment by ghiscoding** [comment]
> Any opinion on the usage question?

I'm not sure what you mean, could you elaborate?

**Comment by 6pac** [comment]
> Both of you: this does raise another usage question. If we are ordering by a column and it is then hidden, should it be removed from the sort list? I can see use cases either way. Should we create an option, say removeHiddenColsFromSort, defaulting to false so as to be non breaking? This would require additional changes, which I can do.

**Comment by ghiscoding** [comment]
I was ok with your PR approach, I would find that weird that we would still sort by a column that is not visible to the user since it could bring too much confusion (I like the approach of "What you see is what you get")

**Comment by 6pac** [comment]
OK, no worries, but in that case I should also filter for columns that exist but are ```hidden```. Will tweak it some more.

Thanks for the hint on debugging too.

**Comment by ghiscoding** [comment]
should be fixed in [v5.9.2](https://github.com/6pac/SlickGrid/releases/tag/5.9.2)

**Comment by LordGalahad** [comment]
Yup it is fixed. Thanks
